build/build-iso.sh

```diff
@@ -2,7 +2,16 @@
 
 # Build Distro ISO using AUTO.ISO minimal auto-install as bootstrap to merge codebase, recompile system, and export ISO
 
-# Run this script inside build/ directory
+# make sure we are in the correct directory
+SCRIPT_DIR=$(realpath "$(dirname "$0")")
+SCRIPT_NAME=$(basename "$0")
+EXPECTED_DIR=$(realpath "$PWD")
+
+if test "${EXPECTED_DIR}" != "${SCRIPT_DIR}"
+then
+	( cd "$SCRIPT_DIR" || exit ; "./$SCRIPT_NAME" );
+	exit
+fi
```

Comment:
Same thing as in sync.sh, you can store the previous PWD:
```sh
PREV_PWD="$PWD"
if [ "${EXPECTED_DIR}" != "${SCRIPT_DIR}" ]; then
	cd "$SCRIPT_DIR" && "./$SCRIPT_NAME"
	cd "$PREV_DIR"
	exit
fi
```

Reply:
Same as https://github.com/Zeal-Operating-System/ZealOS/pull/57#discussion_r861807374.